x-pack/plugins/fleet/server/services/agent_policy.ts

```diff
@@ -629,14 +562,6 @@ class AgentPolicyService {
       throw new HostedAgentPolicyRestrictionRelatedError(`Cannot delete hosted agent policy ${id}`);
     }
 
-    if (agentPolicy.is_default) {
-      throw new Error('The default agent policy cannot be deleted');
-    }
-
-    if (agentPolicy.is_default_fleet_server) {
-      throw new Error('The default fleet server agent policy cannot be deleted');
-    }
-
```

Comment:
Maybe we should prevent agent policies from being deleted if they contain a `fleet_server` package policy in order to make this more explicit. Users would have to remove that package policy first, then delete the agent policy.

A better UX would probably to show a specific confirmation modal when deleting an agent policy with the fleet_server package policy. My concern is that users may accidentally break their whole Fleet setup. @mostlyjason WDYT?

Reply:
A confirmation modal sounds reasonable to explain the impact of deleting it. WDYT @dborodyansky ?

Reply:
Confirmation modal sounds appropriate. See style and writing guidelines for confirming destructive actions here: https://elastic.github.io/eui/#/guidelines/writing/examples#modals

Reply:
@mostlyjason @dborodyansky  thanks, I added this text, let me know if there are any suggestions to change it:
![image](https://user-images.githubusercontent.com/90178898/149323867-f0d99186-008c-48fe-b359-addc2edf766a.png)